File: server/Models/Vault.cs
using System.ComponentModel.DataAnnotations;
using keepr.interfaces;

namespace keepr.Models;

public class Vault : IRepoItem<int>
{
    public int Id { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
    public string Name { get; set; }
    [MaxLength(1000)]
    public string Description { get; set; }
    [MaxLength(1000)]
    public string Img { get; set; }
    public bool IsPrivate { get; set; }
    public string CreatorId { get; set; }
    public Profile Creator { get; set; }
}

public class VaultCreationDTO
{
    public string Name { get; set; }
    [MaxLength(1000)]
    public string Description { get; set; }
    [MaxLength(1000)]
    public string Img { get; set; }
    public bool IsPrivate { get; set; }
}

File: client/src/models/Vault.js
import { Profile } from "./Profile.js"

export class Vault{
    constructor(data){
        this.id = data.id
        this.img = data.img
        this.name = data.name
        this.isPrivate = data.isPrivate
        this.description = data.description
        this.creatorId = data.creatorId
        this.createdAt = new Date(data.createdAt)
        this.updatedAt = new Date(data.updatedAt)
        this.creator = data.creator ? new Profile(data.creator) : null
    }
}